### PR TITLE
build/ops: rpm: stop install-deps.sh clobbering spec file Python build setting

### PR DESCRIPTION
**Pull request by smithfarm**
Fedora 29 still ships a Python 2 binary, but some of Ceph's build
dependencies are only available in py3 versions there. In other
words, from F29 on, it is no longer possible to do a py2 Ceph build
on Fedora, even if a python2 binary exists on the system.

If that were not enough, the Python 2 that ships with Fedora 29 is
linked against a non-compatible version of OpenSSL.

Before this commit, install-deps.sh was overriding the spec file's
Python build setting based on the presence or absence of a python2
binary. As the bug cited below indicates, this was not a good idea.

It's better for the spec file to be explicit about which OS versions
are py2 and which are py3, and just stick to that.

Fixes: http://tracker.ceph.com/issues/37301
Signed-off-by: Brad Hubbard <bhubbard@redhat.com>
Signed-off-by: Nathan Cutler <ncutler@suse.com>

- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: ceph.spec.in [hunk 1]
```diff
@@ -50,11 +50,11 @@
 %endif
 %endif
 %bcond_with seastar
-%if 0%{?fedora} || 0%{?suse_version} >= 1500
-# distros where py2 is _optional_
+%if 0%{?fedora} >= 29 || 0%{?suse_version} >= 1500
```
Comment (tchaikov) [inline ceph.spec.in]: @smithfarm could you amend the commit message with the one at https://github.com/ceph/ceph/commit/6639ffe3a18ebda282515580544e636e45dd20cd ?

i also ran into the OpenSSL issue when running cmake on fc29.

Comment (smithfarm) [reply ceph.spec.in]: Done - thanks.

**Comment by smithfarm** [comment]
Changelog:

* point to bcond documentation in the spec file
* mention that F29 Python 2 is linked against incompatible OpenSSL

File: ceph.spec.in [hunk 2]
```diff
@@ -14,6 +14,13 @@
 #
 # Please submit bugfixes or comments via http://tracker.ceph.com/
 #
+
+#################################################################################
+# conditional build section
```
Comment (dillaman) [inline ceph.spec.in]: Nit: what's with this section header? It seems like something that is really trying to grab someone's attention for a critical concern. I think your comment update on line 61 effectively clears up the confusion re: the syntax.

Comment (badone) [reply ceph.spec.in]: I like it. I think it puts the relevant info at your fingertips when wrestling with these conditional variables. Could've used it yesterday! :)

**Comment by smithfarm** [comment]
> I did not realize that install-deps.sh was doing that level of surgery on the .spec file. Ouch :)

When we first started doing the py3 builds on certain platforms, folks running those platforms wanted a way to get a py2 build. That's why this little bit of "surgery" was added.

Now we are a bit further along and py3 no longer seems so exotic, I guess. Not to mention the py2 build being broken on F29 :-/

**Comment by smithfarm** [comment]
```
Reviewed-by: Jason Dillaman <dillaman@redhat.com>
Reviewed-by: Lenz Grimmer <lgrimmer@suse.com>
Reviewed-by: Kefu Chai <kchai@redhat.com>
Reviewed-by: Ken Dreyer <kdreyer@redhat.com>
Reviewed-by: Brad Hubbard <bhubbard@redhat.com>
Reviewed-by: Tim Serong <tserong@suse.com>
```

**Comment by smithfarm** [comment]
Also, see https://github.com/ceph/ceph/pull/25205 for a related fix.

**Pull request merged**